"Model Workflows" section in Guides

validbeck opened the pull request:
## Internal Notes for Reviewers

| Model Workflows | Details |
|---|---|
| <img width="190" alt="Screenshot 2024-05-31 at 1 07 35 PM" src="https://github.com/validmind/documentation/assets/164545837/ac95c7e3-3c57-4f44-abf6-47752930ff44">| For sc-3870, I created three (3) net-new docs in a `Model Workflows` section under **Guides** to serve as a framework for our workflows feature. | 

### Working with model workflows
- Should say `Working with workflows` in the side nav
- Links out to `Customize resource statuses` and `Set up model workflows` under "What's next"
- Covers reviewing existing workflows attached to models & model documentation, as well as working with workflow-related actions
- Has demonstrative gifs

> See PDF: [Working with model workflows.pdf](https://github.com/user-attachments/files/15518425/Working.with.model.workflows.pdf)

### Customize resource statuses
- Covers the setup for statuses used in workflows

> See PDF: [Customize resource statuses.pdf](https://github.com/user-attachments/files/15518427/Customize.resource.statuses.pdf)

### Set up model workflows
- Should say `Set up workflows` in the side nav
- Covers adding/editing/deleting workflow steps and the details of configuration, including conditional requirements
- Has demonstrative gifs

> See PDF: [Set up model workflows.pdf](https://github.com/user-attachments/files/15518428/Set.up.model.workflows.pdf)

### Guides
Now has a `Model workflows` section.
<img width="972" alt="Screenshot 2024-05-31 at 1 25 31 PM" src="https://github.com/validmind/documentation/assets/164545837/0c671858-0c2c-4567-8827-3577719a15c1">

## External Release Notes

![see-workflow](https://github.com/validmind/documentation/assets/164545837/e2f93d8a-25ba-4155-afbc-16f4416a9528)

You can now manage lifecycle processes within your ValidMind Platform UI setup using workflows. 

Check out our **[new documentation on working with your model workflows!](../guide/working-with-model-workflows.qmd)**

To use workflows, you'll need to:
1. [Customize your resource statuses](../guide/customize-resource-statuses.qmd),
2. [Then set up your model workflows](../guide/set-up-model-workflows.qmd)

nrichers commented on site/guide/set-up-model-workflows.qmd:
```diff
@@ -0,0 +1,213 @@
+::: {.g-col-5}
+![](user-action-custom-field.png){fig-alt="A screenshot showing the action transition panel with the Date On Hold custom field."}
+
+::: {.callout-tip title="Example" appearance="minimal"}
```
FYI, Quarto also supports putting callouts into the right margin which might help clean up a visually very busy section in the docs:

::: {.column-margin}
EXISTING_CALLOUT
:::

validbeck replied:
Interesting! I will try that out and see how it looks.

validbeck replied:
That doesn't look as good for these sets of images, so I'm going with the old version.

nrichers commented on site/guide/set-up-model-workflows.qmd:
```diff
@@ -0,0 +1,213 @@
+- Requires a User type [custom field](manage-inventory-custom-fields.qmd) to select from, and must be connected to both an approval and rejection **[Status Change](#status-change)** step.
+
+::: callout 
+Please note that users must be assigned to the Approval Group custom field before it is transitioned into the **Approval** workflow step in order for the message to be displayed to them. 
```
```suggestion
Please note that you must assign users to the **Approval Group** custom field before it is transitioned into the **Approval** workflow step in order for the message to be displayed to them. 
```

Use active voice, George Orwell said so.

nrichers commented on site/guide/set-up-model-workflows.qmd:
```diff
@@ -0,0 +1,213 @@
+    After you've configured a step, you can then [link your workflow together](#link-workflow-together).
+
+::: {.callout-tip appearance="minimal"}
+Refer to **[Available step types](#available-step-types)** for information on available step types and details on how to set up conditional requirements.
```
GLOBAL: I would not bold links to subsections. This topic already has a ton of highlighting in it, so whatever extra visual clutter we can omit, we should.

nrichers commented on site/guide/set-up-model-workflows.qmd:
```diff
@@ -0,0 +1,213 @@
+|---:|---|
+| When these conditions are met (optional) | [Add conditional requirements](#add-conditional-requirements) to qualify for this step.  |
+| Set status to (required) | Select the status to transition to. |
+: Status Change step configuration {.hover tbl-colwidths="[40,60]"}
```
```suggestion
: **Status Change** step configuration {.hover tbl-colwidths="[40,60]"}
```

GLOBAL: References to UI elements you interact with should be bolded.

Do you know what the icon library is that we're using? We might be able to add the same to Quarto.

nrichers replied:
Also, does `{.hover tbl-colwidths="[40,60]"}` work for you? It doesn't seem to do anything when I tested locally.

validbeck replied:
> Do you know what the icon library is that we're using? We might be able to add the same to Quarto.

I asked Jeffrey about the icons, they're not from Font Awesome (they're little images from Figma). I figured it was more trouble than it was worth to add images here.

> Also, does {.hover tbl-colwidths="[40,60]"} work for you? It doesn't seem to do anything when I tested locally.

It does, it adjusts the width effectively for me, not sure why it doesn't for you. [This is also how the official Quarto docs tell you to assign table widths.](https://quarto.org/docs/authoring/tables.html#column-widths)

nrichers replied:
> It does, it adjusts the width effectively for me, not sure why it doesn't for you. [This is also how the official Quarto docs tell you to assign table widths.](https://quarto.org/docs/authoring/tables.html#column-widths)

I figured it out — I'd never used `.hover` before but mainly because **I try to avoid tables wherever possible**.
Maybe my comments relates to our use of tables more generally — they can be problematic for screen readers and require special attention to be accessible ([How tables can make content less accessible](https://medium.com/@lizzie.g.bruce/accessibility-and-tables-c233f6b2b823)) 

In this case, I do think tables are helpful but they should only be used to represent data or parameters. Tables should not be used to format or group other content together. And even where a table might make sense, I would use a simple definition list before using a table, as it sidesteps accessibility issues altogether.

validbeck replied:
That makes sense. Would you like me to change these to a list, or can we keep them as tables? I think the information here makes *sense* as a table, as they are key / value pairs, but I understand the argument against them.

nrichers replied:
I'd leave as is, but let's avoid a proliferation of tables as we expand our docs. "Use tables sparingly" would likely be a good addition to our style guide at some point.

nrichers commented on site/guide/set-up-model-workflows.qmd:
```diff
@@ -0,0 +1,213 @@
+:::
+
+#### Approval
+- When your resource reaches the approval step, users in the selected Approval Group will receive your configured message. 
```
Is "Approval Group" a UI element? If so, it should be bolded. If not, it should be lowercased. (I also see this referred to as "Approval group" elsewhere, so some consistency sleuthing might be needed.)

nrichers commented on site/guide/set-up-model-workflows.qmd:
```diff
@@ -0,0 +1,213 @@
+
+#### Approval
+- When your resource reaches the approval step, users in the selected Approval Group will receive your configured message. 
+- Requires a User type [custom field](manage-inventory-custom-fields.qmd) to select from, and must be connected to both an approval and rejection **[Status Change](#status-change)** step.
```
```suggestion
- Requires a **User** type [custom field](manage-inventory-custom-fields.qmd) to select from, and must be connected to both an approval and rejection **[Status Change](#status-change)** step.
```

I think that is the actual field name based on the link, so it should be bolded?

nrichers commented on site/guide/set-up-model-workflows.qmd:
```diff
@@ -0,0 +1,213 @@
+- Requires a User type [custom field](manage-inventory-custom-fields.qmd) to select from, and must be connected to both an approval and rejection **[Status Change](#status-change)** step.
+
+::: callout 
+Please note that users must be assigned to the Approval Group custom field before it is transitioned into the **Approval** workflow step in order for the message to be displayed to them. 
```
GLOBAL: Same here, there more unbolded UI elements here.

validbeck replied:
Yeah, I'm still on the hill of "you can't click on these elements, so they shouldn't be bolded"... 😅

nrichers replied:
Let's chat later today and agree on what to do. My take is that since you are being asked to locate that section in the UI, you ARE interacting with it — highlighting this makes sense. We highlight other, non-UI texts for emphasis already.

validbeck replied:
I made a quick change to the style guide as a suggestion, and edited the these workflow docs to reflect that!

nrichers commented on site/guide/set-up-model-workflows.qmd:
```diff
@@ -0,0 +1,213 @@
+::: {.grid}
+
+::: {.g-col-7}
+![](configure-user-action.png){fig-alt="A screenshot showing a User Action step configuration with a request field of Date On Hold."}
```
The screenshots are really helpful for this rather complex task, but they sometimes show extra stuff that adds visual complexity. E.g. here, I'd remove the extraneous stuff outside the modal/popup itself. 

Related, we should have a chat about borders around images. We don't have a standard for this and sometimes I've been adding them manually and sometimes with some CSS. In this PR, images don't have borders but they might benefit from them — ideally we sort this out globally via `styles.css` so they work for all images without having to edit them individually (other than removing manually added borders from existing images).

validbeck replied:
I did already crop this screenshot, and I don't think making it smaller helps here — as a user, I'd expect to see the field in context with the other stuff so I could potentially locate it instead of looking for an element in isolation.

nrichers replied:
It's stuff like this that I am talking about around the edges which doesn't add context, it just looks like the screenshot needs to be edited. I would crop this and other screenshots to be just the modal itself, not the stray background lines or text behind it. 

<img width="536" alt="image" src="https://github.com/validmind/documentation/assets/15148011/02dc4a63-3e5f-4e03-bdeb-266aad32c0a7">

validbeck replied:
Ohhh I see, sorry, I thought you meant the rest of the configuration box. I can definitely make the crop tighter.

nrichers commented on site/guide/set-up-model-workflows.qmd:
```diff
@@ -0,0 +1,213 @@
+
+1. Locate the **Start** of your workflow.
+
+2. Drag from the **.** at the bottom of **Start** to the top **{{< fa circle >}}** on the first step of your workflow.
```
```suggestion
2. Drag from the period (**.**) at the bottom of **Start** to the top **{{< fa circle >}}** on the first step of your workflow.
```

This read like a broken sentence until I clued in. Not ideal, but I wonder if spelling it out isn't preferrable here?

nrichers commented on site/guide/set-up-model-workflows.qmd:
```diff
@@ -0,0 +1,213 @@
+
+6. When you are finished, click **Save workflow** to apply your changes.
+
+#### Link Approval steps
```
```suggestion
#### Link approval steps
```

nrichers reviewed (commented):
I admire your patience with these docs, @validbeck! I left a bunch of minor consistency comments but this PR does a really nice job of explaining complicated and UI-heavy procedural steps. 😮

validbeck commented on site/guide/customize-resource-statuses.qmd:
```diff
@@ -0,0 +1,59 @@
+
+2. In the left sidebar, click [**{{< fa gear >}} Settings**]({{< var vm_url >}}/settings).
+
+3. Under Workplace Settings, select **{{< fa arrow-right-arrow-left >}} Statuses**.
```
I still think the bits of text that aren't clickable shouldn't be bolded, but perhaps we should have a conversation about how to handle those...

validbeck commented on site/_quarto.yml:
```diff
@@ -184,6 +183,14 @@ website:
+          file: guide/working-with-model-workflows.qmd
+          contents:
+          - guide/customize-resource-statuses.qmd
+          - text: "Set up workflows"
```
Yeah, I wanted to make it less crowded, so I actually prefer the shortened link here, but if this is a dealbreaker that's fine.

nrichers replied:
Not a dealbreaker, it just adds something for this one line in the sidebar. It feels like a consistency detail that's being deliberately introduced when it isn't necessary?

nrichers commented:
> Otherwise I think the other cases are us disagreeing on style choices, so perhaps we can talk it over today on our sync?

Works for me! I really mainly just care about consistency.

validbeck commented on site/guide/set-up-model-workflows.qmd:
```diff
@@ -0,0 +1,213 @@
+
+Default workflows are provided for you as suggestions. You can customize the following default workflows:
+
+- **Model documentation:** When documentation is created
+- **Inventory model:** When the model is created
```
I grabbed these descriptions from the UI, so I'm just matching the wording there to minimize confusion.

validbeck commented on site/guide/set-up-model-workflows.qmd:
```diff
@@ -0,0 +1,213 @@
+
+ Setting up your workflow involves two major steps:
+
+ 1. [Configuring workflow steps](#configure-workflow-steps) 
```
Right, but if we match what the heading says, then the `:` doesn't make sense... (the introductory text leading up to it, I mean)

validbeck commented on site/guide/customize-resource-statuses.qmd:
```diff
@@ -0,0 +1,59 @@
+date: last-modified
+---
+
+Statuses can be manipulated via workflow transitions and are used to track the progress of resources through your organization's processes.
```
Apparently we're not happy with the UI copy, because most of this is stolen from that 😅

validbeck commented:
@nrichers, sorry for the back and forth! I decided to make a quick adjustment to our style guide to accommodate for UI elements that aren't clickable:

<img width="992" alt="Screenshot 2024-06-03 at 10 54 00 AM" src="https://github.com/validmind/documentation/assets/164545837/3d7efbd5-2cce-4ee1-9147-f451c2762932">

My rationale here is breaking types of interactions up with different emphasis helps accessibility (as if everything is bolded, that is confusing too), and you're right that these headers and groupings should also be styled differently than the other body text. 

> I've made some adjustments to the "Model Workflows" section with this in mind and will also review for this when I do [sc-4459](https://app.shortcut.com/validmind/story/4459/style-guide-docs-site-audit). Otherwise, I've modified the copy/navigation to something that's a bit more aligned with your suggestions (more or less).

e.g. 
<img width="986" alt="Screenshot 2024-06-03 at 11 30 56 AM" src="https://github.com/validmind/documentation/assets/164545837/396da778-8b84-4c79-bb21-5b3320771a67">

nrichers commented:
> I decided to make a quick adjustment to our style guide to accommodate for UI elements that aren't clickable

I was hoping to have a conversation on this rather than a unilateral declaration. 🙂 But I do agree that clarity here would be helpful.

FWIW, I checked what Apple, Microsoft, and Google do and they're not 100% consistent but bolding is a very common way they suggest for highlighting UI elements. Google is the clearest: 

> When referring to any UI element by name, put its name in bold, using the b element in HTML or ** in Markdown. This includes names for buttons, menus, dialogs, windows, list items, or any other feature on the page that has a visible name.

Note the "any". 

I have to admit, I'm not a huge fan of having a double rule for bolding and italics, especially as it encourages more references to specific UI elements when users should be focusing on the interaction or using a feature. But let's maybe chat at 4 PM rather than trading GitHub comments. 😄

nrichers reviewed (approved):
LGTM (looks GREAT to me)! :shipit: Thank you putting this PR together and for helping us sort out our bolding of UI elements. I also appreciate the cropped screenshots, they do a lot to remove visual clutter. 

For UI elements, our group decision was to bold only elements you interact with directly, e.g. by clicking. Other UI elements will just use plain text, including section headings in a page or UI elements we refer to. This decision is in keeping with your original way of referring to UI elements which is a simpler, less visually cluttered approach.

validbeck commented:
Made some quick changes based on our conversation yesterday:

- Adjusted the style guide for bolding only for specific interaction elements in the UI
- Modified the Workflows docs to reflect this (removed any italicising for headers/grouping elements and just left the casing we see in the UI for wording)
- Quickly removed any secondary links in step-by-steps other than the 1st login to the platform UI step

| Style Guide adjust | Example style edits |
|---|---|
| <img width="997" alt="Screenshot 2024-06-04 at 8 46 27 AM" src="https://github.com/validmind/documentation/assets/164545837/502e218d-2e6b-4fce-adcc-a5dfee29ffea"> | <img width="796" alt="Screenshot 2024-06-04 at 9 03 43 AM" src="https://github.com/validmind/documentation/assets/164545837/0109dd17-ff66-4427-8e0e-2181d02c4057"> |

> - @nrichers I know you already approved but just for visibility. I'm also going to do a quick edit to the top-level comment for external release notes. 
> - Also, do you know if [Manage roles & permissions](https://docs.validmind.ai/guide/manage-roles-and-permissions.html) is something I just don't have access to? Or rather, I feel like the roles/permissions has been shifted in the UI and this guide needs to be updated?

validbeck commented:
> Do you hold the Customer Admin role? If so, you should see the UI options. I've been using the "Jane Doe" user to test while working with ServiceRocket's writer.

I do have the `Customer Admin` role, but I'm saying that the instructions on that page don't line up with what the experience in the UI is like. 😅 

> but we also use backticks (``) for emphasis, e.g. for parameter names that are part of the product. Maybe we should start a story to collect future style guide updates?

This is true, I'll add to the audit story.

panchicore commented:
🍻

nrichers commented:
> I do have the `Customer Admin` role, but I'm saying that the instructions on that page don't line up with what the experience in the UI is like. 😅

I understand now — yes, it looks like the **Roles & Permissions** page has been split up, along with other changes. I'm already chatting with Suba from ServiceRocket and have asked her about updating these docs.
